### PR TITLE
chore(ci): add `data-service-generator` release workflow for hotfix

### PR DESCRIPTION
**Pull request by overbit**
Close: #6908

### Summary
🚑🐳📝

This pull request enhances the release process of the `data-service-generator` package by adding optional dockerization and ECR push features, and by creating a separate workflow for hotfixes. It also improves the changelog and artifact naming conventions.

> _Release the hotfix, unleash the fury_
> _`data-service-generator` is on fire_
> _Dockerize and push, no time to worry_
> _Changelog and artifact, raise them higher_

### Walkthrough
* Add a new workflow file for releasing hotfixes for the data-service-generator package ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-7d6adfee9363fa27dc92d1301524ba29185eebb971a8ee017c26abce1a288f04L1-R20))
* Add a new input parameter to the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0L6-R14))
* Add two new secrets to the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0R20-R23))
* Modify the title of the changelog category in the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0L82-R91))
* Rename the artifact that is uploaded by the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0L112-R153))
* Add several steps to the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0R106-R137))
* Replace the double quotes with single quotes in the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0L71-R80))
* Remove an extra blank line in the workflow template for data-service-generator releases ([link](https://github.com/amplication/amplication/pull/6987/files?diff=unified&w=0#diff-50f2b193dd04f4b1fcd70e5e90fe3d713a09bc019b94ec596a9d4169731440b0L39-R48))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: .github/workflows/release.dsg.template.yml
Comment (overbit) [inline .github/workflows/release.dsg.template.yml:1]: Need to #6986 PR to be merged first

**Review by overbit — COMMENTED** [review]
Need to #6986 PR to be merged first

File: .github/workflows/dsg-release-notes-genertor.yml
Comment (levivannoort) [inline .github/workflows/dsg-release-notes-genertor.yml:1]: file name is spelled incorrect

File: .github/workflows/release.dsg.hotfix.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Release DSG Hotfix
+run-name: Release DSG hotfix (sha:${{ github.sha }})
+on:
```
Comment (levivannoort) [inline .github/workflows/release.dsg.hotfix.yml:3]: The job trigger is on a new-line in all of the other workflows, lets keep it that way for readability

Comment (overbit) [reply .github/workflows/release.dsg.hotfix.yml:3]:
```suggestion

on:
```

**Review by levivannoort — APPROVED** [review]

**Pull request merged**